File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.py
class Solution:
    def maxDistance(self, arrays: List[List[int]]) -> int:
        return reduce(lambda q,b:(max(q[0],b[-1]-q[1],q[2]-b[0]),min(q[1],b[0]),max(q[2],b[-1])),arrays,(0,inf,-inf))[0]

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.java
class Solution {
    public int maxDistance(List<List<Integer>> arrays) {
        int minValue = arrays.get(0).get(0);
        int maxValue = arrays.get(0).get(arrays.get(0).size() - 1);
        int maxDistance = 0;

        for (int i = 1; i < arrays.size(); i++) {
            List<Integer> currentArray = arrays.get(i);

            // Calculate distance using current min and max with other arrays
            maxDistance = Math.max(maxDistance, Math.max(
                Math.abs(currentArray.get(currentArray.size() - 1) - minValue),
                Math.abs(maxValue - currentArray.get(0))
            ));

            // Update the global min and max values
            minValue = Math.min(minValue, currentArray.get(0));
            maxValue = Math.max(maxValue, currentArray.get(currentArray.size() - 1));
        }

        return maxDistance;
    }
}